src/Evolution/DgSubcell/SubcellOptions.hpp

```diff
@@ -0,0 +1,149 @@
+// Distributed under the MIT License.
+// See LICENSE.txt for details.
+
+#pragma once
+
+#include <limits>
+#include <string>
+
+#include "Options/Options.hpp"
+#include "Utilities/TMPL.hpp"
+
+/// \cond
+namespace PUP {
+class er;
+}  // namespace PUP
+/// \endcond
+
+namespace evolution::dg::subcell {
+/*!
+ * \brief Holds the system-agnostic subcell parameters, such as numbers
+ * controlling when to switch between DG and subcell.
+ */
+class SubcellOptions {
```

Comment:
For the parallel linear solvers I found it easier to have a tag for each option instead of such a wrapper class, so I can mix-and-match the tags. What advantages do you see in this wrapper class over individual tags? I have the impression it just adds wrapping code to set up the private attributes, accessor methods and constructors.

Reply:
- fewer tags
- all the system-agnostic parameters are in one location, both in the input file and in the cache/DataBox

I see no advantage to splitting them here. Every DG-subcell evolution needs to specify all these options, no getting around it. There will be additional options that are specific to each executable, and those each executable can choose how to handle (eg multiple tags, classes that hold the parameters, or whatever else one may think of)

Reply:
ok, up to you I think :) I was just wondering if it could be annoying to dealing with changes to these options, e.g. when you want to swap out a TCI. In that case I think it would be easier to make the options separate tags than to template this wrapper class on the TCI(s) in use.

Reply:
There is support for additional TCIs already. The RDMP is a "things are definitely bad now" TCI depending on the parameters, and the Persson needs to be applied to specific variables. Which variables those are depends on the system, but the parameter value should be consistent (and the "best" value is analytically derived anyway). The system can choose to ignore the Persson TCI completely if it wants.

Reply:
Ah I see, so wouldn't it be a nicer interface to place all the TCI options (for these system-agnostic ones and for additional ones) in one group in the input file? Again, up to you :)

Reply:
I don't expect there to be additional options to the TCIs beyond these. Which variable(s) are checked by the Persson TCI will likely be hard-coded, the ability to recover the primitive variables will be hard-coded with options for how to do the primitive recover being specified separately. So in all the cases I can think of and have seen, I don't see there being any additional parameters, and I don't see a reason to try to force the system-agnostic and system-dependent parameters to be grouped into a single thing.

Reply:
ok then.

Reply:
What I could reasonably see as a way to get the best of both worlds would be to have a `group` for `Subcell` with the tags in it, then have the `Tags::SubcellOptions` take all the required option tags that are current defined inside the class and constructs an object that we access from the cache/box. This reduces cache/box tags while keeping things together in the input file. Whether or not that's good/necessary is to be determined once we have some interesting systems working, I think.

Reply:
Ok. A note on reducing number of DataBox tags: I think it's not good if we have to make user-interface (input-file) design decisions based on the argument "we should reduce the number of tags in the DataBox". That sounds like we can easily end up with various workarounds like wrapper classes with different interfaces, just to keep the number of tags low. If the number of tags in the DataBox is really an important factor and doesn't scale to the number of tags we would get in our executables, then that sounds like a severe limitation of the DataBox implementation that we'll have to address. If it isn't, then I think we should do what's best for the code at hand, i.e. use separate tags or a wrapper class where it makes sense. In either case, I think we should make sure that we have a "recommended" way to add options like these for subcell. I think having to write a lot of wrappers like this isn't really a scalable option, since it's a lot of code just to group some tags together.

Reply:
I don't see any benefit to separating the options. In all cases I use any of the options, I use all of the options together (except for the initial data versions). From a code perspective it is _much_ easier to grab all the options at once and then grab them out of the class than having to go and get each one with a separate tag.

In terms of giving something "recommended", yes I agree it's good to have recommendations, but I'd like to wait until we can do the full simulations and have a solid idea of whether or not any additional TCIs or options are even necessary before trying to come up with recommendations.